src/detail/generator_file_reader.cc

```diff
@@ -13,18 +8,105 @@
 #include <caf/error.hpp>
 #include <caf/none.hpp>
 
+#include "broker/config.hh"
 #include "broker/detail/assert.hh"
 #include "broker/detail/generator_file_writer.hh"
 #include "broker/error.hh"
 #include "broker/logger.hh"
 #include "broker/message.hh"
 
-namespace broker {
-namespace detail {
+#ifdef BROKER_WINDOWS
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif // WIN32_LEAN_AND_MEAN
+
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif // NOMINMAX
+
+#include <Windows.h>
+
+namespace {
+
+std::pair<HANDLE, bool> open_file(const char* fname) {
+  auto hdl = CreateFile(fname, GENERIC_READ, 0, nullptr, OPEN_EXISTING,
+                        FILE_FLAG_SEQUENTIAL_SCAN, nullptr);
+  return {hdl, hdl != nullptr};
+}
+
+std::pair<size_t, bool> file_size(HANDLE fd) {
+  LARGE_INTEGER result;
+  if (!GetFileSizeEx(fd, &result))
+    return {0, false};
+  return {static_cast<size_t>(result.QuadPart), true};
+}
+
+void close_file(HANDLE fd) {
+  CloseHandle(fd);
+}
+
+void* memory_map_file(HANDLE fd, size_t) {
+  return CreateFileMapping(fd, nullptr, PAGE_READONLY, 0, 0, nullptr);
+}
 
-generator_file_reader::generator_file_reader(int fd, void* addr,
+void unmap_file(void* addr, size_t) {
+  UnmapViewOfFile(addr);
+}
+
+void* make_file_view(void* mapper, size_t file_size) {
+  return MapViewOfFile(mapper, FILE_MAP_READ, 0, 0, file_size);
+}
+
+} // namespace
+
+#else
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+std::pair<int, bool> open_file(const char* fname) {
```

Comment:
Are the non-windows versions of these functions supposed to be in unnamed-namespace like the windows versions are?

Reply:
Yep, good catch.